// react-multi-form/dist/index.js
function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = require('react');
var React__default = _interopDefault(React);
var styled = require('styled-components');
var styled__default = _interopDefault(styled);

function _taggedTemplateLiteralLoose(strings, raw) {
  if (!raw) {
    raw = strings.slice(0);
  }

  strings.raw = raw;
  return strings;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n    width: 20px;\n    height: 20px;\n    background-color: ", ";\n    \n    &:before {\n      position: absolute;\n      content: '\u2713';\n      color: #fff;\n      top: 50%;\n      left: 50%;\n      transform: translate(-50%, -50%);\n    }\n  "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n    width: 20px;\n    height: 20px;\n    background-color: ", ";\n    \n    &:before {\n      position: absolute;\n      content: '';\n      background-color: #fff;\n      width: 8px;\n      height: 8px;\n      border-radius: 8px;\n      top: 50%;\n      left: 50%;\n      transform: translate(-50%, -50%);\n    }\n  "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n    &:after {\n      position: absolute;\n      content: '", "';\n      width: max-content;\n      color: ", ";\n      font-weight: ", ";\n      text-transform: uppercase;\n      font-size: 11px;\n      bottom: -20px;\n      left: 50%;\n      transform: translateX(-50%);\n    }\n  "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  position: relative;\n  width: 10px;\n  height: 10px;\n  border-radius: 20px;\n  background-color: #999;\n  transition: all 0.4s;\n\n  ", "\n\n  ", "\n\n  ", "\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var Pill = styled__default.div(_templateObject(), function (props) {
  return props.label && styled.css(_templateObject2(), props.label, function (props) {
    return props.active || props.complete ? props.accentColor || '#24a19c' : '#999';
  }, function (props) {
    return props.active ? 600 : 400;
  });
}, function (props) {
  return props.active && styled.css(_templateObject3(), props.accentColor || '#24a19c');
}, function (props) {
  return props.complete && styled.css(_templateObject4(), props.accentColor || '#24a19c');
});

function _templateObject2$1() {
  var data = _taggedTemplateLiteralLoose(["\n    background-color: ", ";\n    height: 2px;\n  "]);

  _templateObject2$1 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject$1() {
  var data = _taggedTemplateLiteralLoose(["\n  height: 1.5px;\n  background-color: #f2f0ed;\n  flex: 1;\n  transition: all 0.4s;\n\n  ", "\n"]);

  _templateObject$1 = function _templateObject() {
    return data;
  };

  return data;
}
var Line = styled__default.div(_templateObject$1(), function (props) {
  return props.complete && styled.css(_templateObject2$1(), function (props) {
    return props.accentColor || '#24a19c';
  });
});

function MultiStepForm(props) {
  return /*#__PURE__*/React__default.createElement("div", {
    style: styles.container
  }, /*#__PURE__*/React__default.createElement("div", {
    style: styles.stepperContainer
  }, React__default.Children.map(props.children, function (child, i) {
    if (i === React.Children.count(props.children) - 1) return /*#__PURE__*/React__default.createElement(React.Fragment, null, /*#__PURE__*/React__default.createElement(Pill, {
      active: props.activeStep === i + 1,
      label: child.props.label,
      complete: props.activeStep > i + 1,
      accentColor: props.accentColor
    }));
    return /*#__PURE__*/React__default.createElement(React.Fragment, null, /*#__PURE__*/React__default.createElement(Pill, {
      active: props.activeStep === i + 1,
      complete: props.activeStep > 1 && i + 1 < props.activeStep,
      label: child.props.label,
      accentColor: props.accentColor
    }), /*#__PURE__*/React__default.createElement(Line, {
      complete: props.activeStep > 1 && i + 1 < props.activeStep,
      accentColor: props.accentColor
    }));
  })), React__default.Children.map(props.children, function (child, i) {
    if (i + 1 === props.activeStep) return child;
    return React.cloneElement(child, {
      hidden: true
    });
  }));
}
var styles = {
  stepperContainer: {
    display: 'flex',
    justifyContent: 'space-evenly',
    alignItems: 'center',
    marginBottom: 40
  }
};

function _templateObject$2() {
  var data = _taggedTemplateLiteralLoose(["\n  display: ", "\n"]);

  _templateObject$2 = function _templateObject() {
    return data;
  };

  return data;
}
var Container = styled__default.div(_templateObject$2(), function (props) {
  return props.hidden ? 'none' : 'block';
});
function Step(props) {
  return /*#__PURE__*/React__default.createElement(Container, props, props.children);
}

exports.MultiStepForm = MultiStepForm;
exports.Step = Step;
